packages/network-controller/src/NetworkController.ts

```diff
@@ -302,11 +319,10 @@ export class NetworkController extends BaseControllerV2<
    * Refreshes the current network code.
    */
   async lookupNetwork() {
-    /* istanbul ignore if */
+    const releaseLock = await this.mutex.acquire();
```

Comment:
Hmm, that's a good point. If I keep the existing change and run the whole test suite, tests that involve this method will fail at the `this.ethQuery.sendAsync` line because `this.ethQuery` is undefined. I think it has something to do with the fact that `lookupNetwork` completes out of band, but I'm not exactly sure why. I'll take a closer look at this.

Reply:
The fact that we aren't waiting for `lookupNetwork` is making things really tricky 😕  I made some changes in [14c0c9a](https://github.com/MetaMask/core/pull/1039/commits/14c0c9aae2fb6574a9c12fe9951b017e61959256). Let me know if they make sense to you.

Reply:
Perhaps we could update `lookupNetwork` ahead of these changes to make it block on the network call

Reply:
Okay, I'll extract this change to a separate PR.

Reply:
I am trying to extract this change to a new PR, but working with the existing tests is really frustrating, because requests to `net_version` as well as block tracker requests seem to be bleeding over into other tests. If I attempt to mock network requests for a single request (i.e. call `nock.disableNetConnect()`, test the behavior, then call `nock.enableNetConnect()`), there are extra requests made out of band that get caught by Nock. I'm not sure exactly what's happening, but my sense is that because these requests are happening "somewhere else", the requests that we mocked inside of the test that spawned them are no longer mocked by the time that it occurs, because Jest has moved on to another test. For some reason, things work better in these rewritten tests, possibly because I'm more particular about waiting for state changes to occur before moving on (or because I'm using a fake provider, so asynchronous actions take a shorter amount of time to complete). In any case we would have to make some changes to the existing tests, I think, in order to be able to extract this.

Reply:
Sorry I'm having trouble following what it was that you were trying to attempt. I thought you were talking about making `lookupNetwork` block on getting the network ID, but that behaviour isn't currently tested at all.

I tried this out in a separate PR and it seems to pass all tests: https://github.com/MetaMask/core/pull/1063